Lib/modulefinder.py

```diff
@@ -93,7 +94,11 @@ def _find_module(name, path=None):
     else:  # Should never happen.
         return None, None, ("", "", _SEARCH_ERROR)
 
-    file = open(file_path, mode)
+    if mode == 'r':
+        file = tokenize.open(file_path)
+    else:
+        file = open(file_path, mode)
+
```

Comment:
another way to implement this that's probably even simpler (as suggested by the other reviewer) is to change the `mode` to `rb` above and change this line to `+ b'\n'`: https://github.com/python/cpython/blob/2b33a8acf27bd2018f6dd6ed6fa404e73859604d/Lib/modulefinder.py#L348

since `compile` can take bytes directly and do the necessary encoding conversion

Reply:
Do we even need the `+'\n'`?

Reply:
probably not? I wonder why it's there :thinking:

Reply:
seems to be from one of the initial commits of python, perhaps when `compile` didn't work for files without ending in a newline -- the `+'\n'` can be removed entirely now probably:

```
$ git log -M3 -G "'\\\\n'" -- Tools/freeze/modulefinder.py
commit 41c554fbec1be5412aea2b388f0952657a2f07e7
Author: Just van Rossum <just@letterror.com>
Date:   Tue Dec 31 16:27:33 2002 +0000

    moving modulefinder.py to the standard library

commit 78fc3634cbfd65a6be8abfd1b7fc7cbd0ccbfb39
Author: Guido van Rossum <guido@python.org>
Date:   Fri Mar 20 17:37:24 1998 +0000

    Big changes by Mark Hammond to support freezing with DLLs on Windows.
```

Reply:
Hey guys, thanks for the feedback! I just committed changes that I believe that implement your suggestions, please check if it's correct!